distribution/lib/Standard/Base/0.0.0-dev/src/Data/Vector.enso

```diff
@@ -613,36 +627,52 @@ type Vector a
 
              [1] + [2]
     + : Vector Any -> Vector Any
-    + self that =
-        self_len = self.length
-        arr = Array.new (self_len + that.length)
-        Array.copy self.to_array 0 arr 0 self_len
-        Array.copy that.to_array 0 arr self_len that.length
-        Vector.from_polyglot_array arr
-
-    ## Add `element` to the beginning of `self` vector.
+    + self that = case that of
+        _ : Vector ->
+            self_len = self.length
+            arr = Array.new (self_len + that.length)
+            Array.copy self.to_array 0 arr 0 self_len
+            Array.copy that.to_array 0 arr self_len that.length
+            Vector.from_polyglot_array arr
+        _ : Array -> self + Vector.from_polyglot_array that
+        _ -> Error.throw (Type_Error.Error Vector that "Should be a Vector or Array to be appended.")
+
+    ## Inserts the given item into the vector at the given index.
 
        Arguments:
-       - `element`: An element to add to this vector.
-
-       > Example
-         Add 1 to the start of the vector.
-
-             [2, 3].prepend 1
-    prepend : Any -> Vector Any
-    prepend self element = [element] + self
-
-    ## Add `element` to the end of `self` vector.
+       - at: The index at which to insert the item.
+         If the index is greater than the length of the vector, the item will
+         be added at the end.
+         If the index is less than 0, the index will be counted back from the
+         end. If before the beginning will be added at the start.
+       - item: The item to insert into the vector.
```

Comment:
It is often unclear if the element is added before or after the current element at the given index. Of course reading carefully it can be understood that now the new element will be placed at the provided index (i.e. it's inserted 'before' current one and current one and all after it are shifted to the right). Still I think to make it super clear for the users, I'd add a note that it inserts the element so that now it will be present at the provided index in the new vector and all the elements following that index from the old vector will be shifted one element to the right. It may be good to also provide some example, like:

```
['a', 'b', 'c'].insert 1 'X' == ['a', 'X', 'b', 'c']
['a', 'b', 'c'].insert item='X' == ['a', 'b', 'c', 'X']
['a', 'b', 'c'].insert -1 'X' == ['a', 'b', 'X', 'c']
```